### PR TITLE
fix: TextArea control with maxLength

### LINKED ISSUE #30066: Input.TextArea的maxLength和countFormatter失效
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/sleepy-drake-nvlj4?file=/src/App.js:368-379)

### Steps to reproduce
通过重现链接可以直接演示操作

### What is expected?
1. 可以限制输入框的输入长度，我应该怎么做？
2. countFormatter props如何正确的使用？

### What is actually happening?
1. 在受控模式下无法限制输入
2.countFormatter按照文档的调用方式不生效

| Environment | Info |
|---|---|
| antd | 4.15.0 |
| React | 17.0.2 |
| System | macOS 11.2.3 (20D91) |
| Browser | 版本 89.0.4389.114（正式版本） (x86_64) |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #30066

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix TextArea `maxLength` not work with control mode.        |
| 🇨🇳 Chinese |   修复 TextArea `maxLength` 在受控时无效的问题。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/index.zh-CN.md [hunk 1]
```diff
@@ -46,7 +46,6 @@ Input 的其他属性和 React 自带的 [input](https://reactjs.org/docs/dom-el
 | allowClear | 可以点击清除图标删除内容 | boolean | false |  |
 | autoSize | 自适应内容高度，可设置为 true \| false 或对象：{ minRows: 2, maxRows: 6 } | boolean \| object | false |  |
 | bordered | 是否有边框 | boolean | true | 4.5.0 |
-| countFormatter | 指定字数展示的格式 | (count: number, maxLength?: number) => string | - | 4.10.0 |
```
Comment (hengkx) [inline components/input/index.zh-CN.md:49]: 怎么移除了api?

Comment (zombieJ) [reply components/input/index.zh-CN.md:49]: #30066 里问了就看了一下历史，发现这个 API 从来没有过。

Comment (yoyo837) [reply components/input/index.zh-CN.md:49]: #28145 @MrHeer 看看有印象吗？

Comment (zombieJ) [reply components/input/index.zh-CN.md:49]: 先合了，讨论可以继续~

Comment (MrHeer) [reply components/input/index.zh-CN.md:49]: 我的锅，最开始我添加的是 `countFormatter`, 后面 @afc163 大佬让我改了 API，中文文档忘记删了。

**Comment by MrHeer** [comment]
PR 的标题 `maxLenght` 打错了 `maxLength`

**Pull request merged**